### PR TITLE
GUVNOR-2822: [Guided Decision Table] Moving back multiple patterns in wizard throws exception

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/GUVNOR-2822

guvnor-2822

### CODE REVIEW

**Comment by karreiro** [comment]
@manstis @jomarko Could you please take a look?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/table/pages/FactPatternsPageViewImpl.java [hunk 1]
```diff
@@ -149,275 +146,272 @@ public void setup() {
...
         presenter.stateChanged();
     }
 
     @UiHandler(value = "btnRemove")
-    public void btnRemoveClick( final ClickEvent event ) {
+    public void btnRemoveClick(final ClickEvent event) {
         boolean allPatternsRemoved = true;
-        for ( Pattern52 pattern : chosenPatternSelections ) {
-            if ( !validator.canPatternBeRemoved( pattern ) ) {
+        for (final Pattern52 pattern : new ArrayList<Pattern52>(chosenPatternSelections)) {
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/table/pages/FactPatternsPageViewImpl.java]: The real change is here.

Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/table/pages/FactPatternsPageViewImpl.java]: @karreiro could you please tell me more about this change? I mean I am not sure why is needed to wrap `chosenPatternSelections` into new array list.

Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/table/pages/FactPatternsPageViewImpl.java]: @jomarko I'm guessing the old code threw a "Concurrent Modification Exception" but let @karreiro confirm.

Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/table/pages/FactPatternsPageViewImpl.java]: You guessed it right, @manstis! ;-)

**Pull request merged**